### PR TITLE
add cvm and other tests

### PR DESCRIPTION
**Pull request by agokarn**

### CODE REVIEW

File: ohttp-server/src/main.rs [hunk 1]
```diff
@@ -630,52 +653,370 @@ mod general_purpose_tests {
...
+            let chunk = std::str::from_utf8(&chunk).expect("Could not get chunk");
+            info!("{chunk}");
+        }
+
+        shutdown_server(server_handle, shutdown_channel_sender).await;
+    }
+
+    #[tokio::test]
+    // Invalid discover endpoint for local testing
+    async fn non_cvm_fail1() {
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:761]: can we give more descriptive names for what this is trying to test?

File: ohttp-server/src/main.rs [hunk 2]
```diff
@@ -630,52 +653,370 @@ mod general_purpose_tests {
...
+        } else {
+            error!("Failed to find the file at: {:?}", path);
+            return None;
+        }
+
+        Some(path)
+    }
+
+    #[tokio::test]
+    async fn cvm_test_basic() {
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:910]: again CVM is perhaps not the right name, perhaps kms_?

File: ohttp-server/src/main.rs [hunk 3]
```diff
@@ -630,52 +653,370 @@ mod general_purpose_tests {
...
+            panic!("This should never happen!")
+        }
+    }
+
+    const KMS_URL_SERVER_DEBUG: &str =
+        "https://accconfinferencedebug.confidential-ledger.azure.com/app/key";
+
+    #[tokio::test]
+    // mismatched KMS for client and server
+    async fn cvm_test_fail2() {
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:984]: This should be called mismatched_kms_test

File: ohttp-server/src/main.rs [hunk 4]
```diff
@@ -630,52 +653,392 @@ mod general_purpose_tests {
...
+            .get("https://accconfinferenceprod.confidential-ledger.azure.com/node/network")
+            .send()
+            .await
+            .expect("reqwest::Client::get() failed");
+
+        let body: Value = response.json().await.expect("Failed to read response");
+        info!("body = {:?}", body);
+
+        // Accessing the "name" field
+        if let Value::Map(map) = body {
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:895]: Wonder if its possible to write this big nested if better, perhaps for a later PR?

**Review by kapilvgit — APPROVED** [review]

**Review by yf23 — APPROVED** [review]

**Pull request merged**